index.rst

```diff
@@ -1,51 +1,1667 @@
+
+and the line
+
+.. code-block:: python
+
+    scripts.BasicSConscript.tests(pyList=pybind11_ported_tests)
+
+must be changed to
+
+.. code-block:: python
+
+    scripts.BasicSConscript.tests()
+
+.. note::
+
+    It is possible that scripts.BasicSConscript.tests might contain other args or kwargs,
+    in which case ``pyList=pybind11_ported_tests`` is inserted as a new kwarg.
+
+Running all Tests
+=================
+
+Before merging a test with the main branch ``DM-8467`` you should always ensure that all 
+of the tests wrapped with pybind11, not just the new ones wrapped in the current branch, still succeed. 
+There is a text file ``tests/test.txt`` that lists all of the tests in the current package.
+To run all of the wrapped tests use:
+
+.. code::
+
+    py.test `sed -e '/^#/d' tests/test.txt`
+
+.. _new_test:
+
+Wrapping a New Test
+===================
+
+Setup
+-----
+
+Since the stack has been built using the pybind11 branch of lsstsw,
+once lsstsw has been setup you can simply use
+
+.. code-block:: bash
+
+    cd <repository directory>
+    setup -r .
+
+to setup the package currently being wrapped.
+
+.. _locking:
+
+Rebasing
+--------
+
+Because the pybind11 stack is a fork of the master lsst packages,
+frequent rebasing will occur throughout the pybind11 port.
+Additionally, while we strive to have different developers work as much as possible on independent packages,
+the numerous interdependencies will sometimes require working on the same package and even in the same 
+ticket branch. Thus frequent pushing and rebasing is necessary to keep everyone's stack up to date.
+To rebase from the current pybind11 master, DM-8467, use
+
+.. code-block:: bash
+
+    $ checkout tickets/DM-8467
+    $ git fetch
+    $ git reset --hard origin/tickets/DM-8467
+    $ git checkout <branch>
+    $ git rebase --onto tickets/DM-8467 C~ tickets/<branch>
+
+where ``<branch>`` is the branch to update and ``C`` is the first commit made in the current ticket.
+This series of commands does a force pull to get the latest version of DM-8467 and then rebases all of the
+new commits on top of the rebased DM-8467.
+
+Building the current test
+-------------------------
+
+As you wrap the package it can be useful to compile the package using
+
+.. code-block:: bash
+
+    $ scons python lib
+
+which only builds the changes to the package and does not run any of the tests,
+which can save a substantial amount of time.
+
+.. _activate_test:
+
+Activating and skipping tests
+-----------------------------
+
+Many test files have multiple tests and sometimes even multiple test classes inside of them.
+It can be useful to only run one test at a time (to prevent a bombardment of errors),
+so the decorator ``@unittest.skip("TODO:pybind11")`` should be added to all of the tests and
+classes other than the one you are currently wrapping.
+Also make sure to uncomment the test in ``tests/test.txt`` so that the test will be run by scons.
```

Comment:
JIRA isn't currently running pytest is it? Even if it is there has to be some place to tell 
pytest which tests to run, and it is better to have a text file with all of the tests (and unused tests commented out) than to add tests as they are wrapped. Otherwise you get a large number of merge conflicts when rebasing.